hack/run-integration-tests.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BINDIR="${SCRIPTDIR}/../out/bin"
+KREW_BINARY_DEFAULT="$BINDIR/krew-linux_amd64"
+
+if [[ "$#" -gt 0 && ( "$1" = '-h' || "$1" = '--help' ) ]]; then
+  cat <<EOF
+Usage:
+  $0 krew  # uses the given krew binary for running integration tests
+  $0       # assumes a pre-built krew binary at $KREW_BINARY_DEFAULT
+EOF
+  exit 0
+fi
+
+install_kubectl_if_needed() {
```

Comment:
Big -1 for this. It's not clear from the name that this script installs kubectl –so let's not do that.

You don't want to do things like this on dev machines either. For travis CI we should just export this as a step there.

Reply:
I think that concern is a bit exaggerated. After all, the binary gets placed in the `out/` directory which is to be considered ephemeral. However, I can make the downloading more visible and only enable it when passing a flag like `--download-kubectl-if-required` or so.

I would like to avoid extra preparatory steps in travis, that's why I put everything in that script. That also makes passing in the env variables more convenient which otherwise would clutter the travis steps.

Reply:
Regardless I think it should be decoupled. 

e.g. travis can do: env KUBECTL=out/bin/kubectl hack/get-kubectl.sh
and in the tests we offer a $KUBECTL customization. 
This way the integration tests have no idea about what to do when kubectl is missing. The hidden/auto magic behavior tends to trip people occasionally. So let’s stick to what the script says it does in the filename.

Reply:
Sorry for my slow response on this topic. If you have such a strong opinion here, then let's do it your way.

As there is a separate issue for this, I'll leave that for another PR though.

Reply:
I opened #210 for  this.